shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -107,3 +130,133 @@
+// are handled by the embedded view they are not delivered to the Flutter framework in this phase
+// as well. So during this phase as well the ForwardingGestureRecognizer dispatched the events
+// directly to the FlutterView.
+@interface ForwardingGestureRecognizer : UIGestureRecognizer <UIGestureRecognizerDelegate>
+- (instancetype)initWithTarget:(id)target flutterView:(UIView*)flutterView;
+@end
+
+@implementation TouchInterceptingView {
+  ForwardingGestureRecognizer* forwardingRecognizer;
+  DelayingGestureRecognizer* delayingRecognizer;
+}
+- (instancetype)initWithSubView:(UIView*)embeddedView flutterView:(UIView*)flutterView {
+  self = [super initWithFrame:embeddedView.frame];
+  if (self) {
+    self.multipleTouchEnabled = YES;
+    embeddedView.autoresizingMask =
+        (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+
+    [self addSubview:embeddedView];
+
+    forwardingRecognizer =
+        [[[ForwardingGestureRecognizer alloc] initWithTarget:self
+                                                 flutterView:flutterView] autorelease];
+
+    delayingRecognizer = [[[DelayingGestureRecognizer alloc] initWithTarget:self
+                                                                     action:nil] autorelease];
+
+    [self addGestureRecognizer:delayingRecognizer];
+    [self addGestureRecognizer:forwardingRecognizer];
+  }
+  return self;
+}
+
+- (void)releaseGesture {
+  delayingRecognizer.state = UIGestureRecognizerStateFailed;
+}
+@end
+
+@implementation DelayingGestureRecognizer
+- (instancetype)initWithTarget:(id)target action:(SEL)action {
+  self = [super initWithTarget:target action:action];
+  if (self) {
+    self.delaysTouchesBegan = YES;
+    self.delegate = self;
+  }
+  return self;
+}
+
+- (BOOL)gestureRecognizer:(UIGestureRecognizer*)gestureRecognizer
+    shouldBeRequiredToFailByGestureRecognizer:(UIGestureRecognizer*)otherGestureRecognizer {
+  return otherGestureRecognizer != self;
+}
+
+- (BOOL)gestureRecognizer:(UIGestureRecognizer*)gestureRecognizer
+    shouldRequireFailureOfGestureRecognizer:(UIGestureRecognizer*)otherGestureRecognizer {
+  return otherGestureRecognizer == self;
+}
+
+- (void)touchesEnded:(NSSet*)touches withEvent:(UIEvent*)event {
+  // The gesture has ended, and the delaying gesture recognizer was not failed, we recognize
+  // the gesture to prevent the touches from being dispatched to the embedded view.
+  //
+  // This doesn't work well with gestures that are recognized by the Flutter framework after
+  // all pointers are up.
+  //
+  // TODO(amirh): explore if we can instead set this to recognized when the next touch sequence
+  //  begins, or we can use a framework signal for restarting the recognizers (e.g when the
+  //  gesture arena is resolved).
+  self.state = UIGestureRecognizerStateRecognized;
+}
+
+- (void)touchesCancelled:(NSSet*)touches withEvent:(UIEvent*)event {
+  self.state = UIGestureRecognizerStateCancelled;
```

Comment:
This is an illegal recognizer. You cannot go to the cancelled state without beginning a recognition. Please follow all the rules of either [continuous or discrete gesture recognition](https://developer.apple.com/documentation/uikit/touches_presses_and_gestures/implementing_a_custom_gesture_recognizer/about_the_gesture_recognizer_state_machine?language=objc). This will allow these recognizers to interact correctly with other recognizers on the system.

Reply:
I think you need continuous recognizers here BTW.

Reply:
Changed it to recognized in this case.